source/MaterialXGenShader/OCIOColorManagementSystem.cpp

```diff
@@ -222,11 +398,18 @@ ImplementationPtr OCIOColorManagementSystem::getImplementation(const ColorSpaceT
         // a valid function name is created here
         std::string transformFunctionName = "IM_" + createValidName(transform.sourceSpace) + "_to_" + createValidName(transform.targetSpace) + "_" + typeName + "_ocio";
```

Comment:
Note: `createValidName()` must forbid white spaces, and double underscores.

Reply:
This is done in 2 parts:
1. `createValidname()` will remove any spaces and other invalid characters. 
2. Things like double underscores is handled by each code generator as it's language specific -- e.g. for the GLSL generator double underscores are "sanitized" during code generation but this would not occur for the OSL generator for example.